### PR TITLE
Show_all_users

### PR DESCRIPTION
**Pull request by samnangcattor**

### CODE REVIEW

**Comment by RathanakSreang** [comment]
Ok

File: Gemfile [hunk 1]
```diff
@@ -6,7 +6,10 @@ gem "rails", "4.2.1"
 # Use sqlite3 as the database for Active Record
 gem "mysql2"
 gem "devise"
+gem "bcrypt", "3.1.7"
```
Comment (RathanakSreang) [inline Gemfile]: remove this gem

**Comment by HorNorin** [comment]
ok

**Comment by RathanakSreang** [comment]
Ok

File: app/views/admins/users/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,6 @@
+<% provide :title, t("titles.user") %>
+<h1><%= t "title.user" %></h1>
+<ul class = "users">
+  <%= render partial: "admins/users/user", collection: @users, as: :user %>
```
Comment (chuanhtuan) [inline app/views/admins/users/index.html.erb]: <%= render collection: @users, as: :user %>
em thử cách trên xem có chạy không nhé

File: app/controllers/admins/sessions_controller.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+class Admins:SessionsController < Desive::SessionsController
+  def new
+    super
+  end
+
+  def create
+    user = User.find_by_email params[:admin][:email]
```
Comment (norifumi) [inline app/controllers/admins/sessions_controller.rb]: You should not use `find_by_xxxx` methods. They are deprecated from Rails 4.
You should use `bind_by` method as like as `find_by email:  params[:admin][:email]` instead.

File: app/controllers/admins/sessions_controller.rb [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+    if user && user.admin?
+      super
+    else
+      flash[:danger] = t("flashs.login")
+      redirect_to new_admin_session_path
+    end
```
Comment (norifumi) [inline app/controllers/admins/sessions_controller.rb]: Do you know guard clause?
It is like as
```ruby
  unless user && user.admin?
    redirect_to new_admin_session_path, flash: {danger: t("flashs.login")}
  end
  super
```
It doesn't have nested conditions, but simply guard clause are more readable.

File: app/views/admins/users/index.html.erb (hunk 2 above)
Comment (samnangcattor) [inline app/views/admins/users/index.html.erb]: không  chạy đâu anh ơi nhưng  em có thể viết  ngắn hơn là  
<%= render @users %>

File: db/seeds.rb [hunk 5]
```diff
@@ -5,3 +5,11 @@
 #
 #   cities = City.create([{ name: 'Chicago' }, { name: 'Copenhagen' }])
 #   Mayor.create(name: 'Emanuel', city: cities.first)
+50.times do
```
Comment (norifumi) [inline db/seeds.rb]: Seeds should be used for master datas. When you want to create datas for test, you create rake task or data factory tool e.g. FactoryGirl https://github.com/thoughtbot/factory_girl .

Comment (chuanhtuan) [inline db/seeds.rb]: I'm sorry, I teach them to use seed file for data test
I will teach them using factory_girl
Thanks!

Comment (khanhkhanh) [inline db/seeds.rb]: @norifumi @chuanhtuan Don't worry about that, @dieunb told us to use FactoryGirl. You should allow us to use seeds first (for example in 1st project) before using Factory_Girl. It's a nice way for us to grow up :)

Comment (norifumi) [inline db/seeds.rb]: OK

File: app/controllers/admins/sessions_controller.rb [hunk 6]
```diff
@@ -0,0 +1,12 @@
+class Admins:SessionsController < Desive::SessionsController
+  def new
+    super
+  end
+
+  def create
+    user = User.find_by params[:admin][:email]
```
Comment (norifumi) [inline app/controllers/admins/sessions_controller.rb]: You intent to find by email, didn't you?
```ruby
user = User.find_by email: params[:admin][:email]
```

**Comment by norifumi** [comment]
LGTM, OK

**Pull request merged**